internal/pkg/term/syncbuffer/syncbuffer_test.go

```diff
@@ -0,0 +1,167 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package syncbuffer
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestSyncBuffer_Write(t *testing.T) {
+	testCases := map[string]struct {
+		input        []byte
+		wantedOutput string
+	}{
+		"append to custom buffer with simple input": {
+			input:        []byte("hello world"),
+			wantedOutput: "hello world",
+		},
+		"append to custom buffer with empty input": {
+			input:        []byte(""),
+			wantedOutput: "",
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			// GIVEN
+			sb := &SyncBuffer{}
+
+			// WHEN
+			sb.Write(tc.input)
+
+			// THEN
+			require.Equal(t, tc.wantedOutput, sb.buf.String())
+		})
+	}
+}
+
+func TestSyncBuffer_IsDone(t *testing.T) {
+	testCases := map[string]struct {
+		buffer     *SyncBuffer
+		wantedDone bool
+	}{
+		"Buffer is done": {
+			buffer:     &SyncBuffer{done: make(chan struct{}), buf: bytes.Buffer{}},
```

Comment:
consider always using _only_ the public API of a package (i.e. don't set any private fields or check against private fields in unit tests).

This is because these fields are internal details, we should be able to change the guts of a struct without having to update the unit tests. In this situation, you'd do:
```suggestion
			buffer:     New(),
```